credmon/CredentialMonitors/OAuthCredmonWebserver/OAuthCredmonWebserver.py

```diff
@@ -180,8 +181,12 @@ def oauth_login(provider):
     Go to OAuth provider
     """
 
-    if not (provider in session['providers']):
-        raise Exception("Provider {0} not in list of providers".format(provider))
+    try:
```

Comment:
Catching a `KeyError` is strange.  Simply have two `if` statements, the first one checking if `'providers'` is in `sessions`.

Reply:
Does https://github.com/htcondor/scitokens-credmon/pull/21/commits/59b93e22fe4889bf7d22d8dadb72a2e891f4ab2d address this alright?